bootloader/grub.go

```diff
@@ -317,6 +317,19 @@ func (g *grub) TryKernel() (snap.PlaceInfo, error) {
 	return nil, ErrNoTryKernelRef
 }
 
+// UpdateBootConfig updates the grub boot config only if it is already managed
+// and has a lower edition.
+//
+// Implements ManagedAssetsBootloader for the grub bootloader.
+func (g *grub) UpdateBootConfig(opts *Options) error {
+	bootScriptName := "grub.cfg"
+	currentBootScript := filepath.Join(g.dir(), "grub.cfg")
+	if opts != nil && opts.Recovery {
```

Comment:
Is the intent here to replace boot script with recovery boot script when `opts.Recovery` is set? I would have expected an update to the recovery boot to be symmetric. Perhaps I'm missing something. In either case this could use a code comment.